code/test/Fakes/MSBuild/FakeMsBuildProject.cs

```diff
@@ -112,30 +112,32 @@ private bool ItemExists(string itemPath)
 
         private VsItemType GetItemType(string fileName)
         {
-            string ext = Path.GetExtension(fileName).ToLower();
-            if (ext == ".cs")
-            {
-                if (fileName.ToLower().Contains(".xaml.cs"))
-                {
-                    return VsItemType.CompiledWithDependant;
-                }
-                else
-                {
-                    return VsItemType.Compiled;
-                }
-            }
-            else if (ext == ".xaml")
-            {
-                return VsItemType.XamlPage;
-            }
-            else if (ext == ".resw")
-            {
-                return VsItemType.Resource;
-            }
-            else
+            VsItemType returnType = VsItemType.Content;
+
+            switch (Path.GetExtension(fileName).ToLower())
             {
-                return VsItemType.Content;
+                case ".cs":
+                    if (fileName.ToLower().EndsWith(".xaml.cs"))
```

Comment:
you can't really do invariant comparison in the switch cases, unless you use the latest C# features, but in that case I think tolower up front for reuse in all the cases is actually better